File: leetcode/easy/palindrome-number.py
def isPalindrome(x):
    rev = 0

    if x < 0 or (x % 10 == 0 and x != 0):
        return False

    while(x > rev):
        digit = x % 10
        rev = rev * 10 + digit
        x = x // 10

    if x == rev or x == (rev // 10):
        return True

    return False

print(isPalindrome(11))

File: leetcode/easy/palindrome-number.php
<?php
function isPalindrome($x) {
  $rev = 0;

  if ($x < 0 || ($x % 10 == 0 && $x != 0)) {
    return false;
  }

  while($x > $rev){
    $digit = $x % 10;
    $rev = $rev * 10 + $digit;
    $x = (int)$x / 10;
  }

  if ($x == $rev || ($x == (int)$rev/10)) {
    return true;
  }

  return false;
}

var_dump(isPalindrome(10));
